### PR TITLE
Add logging to cleanup method and use profile.activated

### PR DESCRIPTION
**Pull request by joshblum**
Addresses #233. @felixSchl sorry this so long to get around to. let me know if this addresses your issue! @dicato wasn't sure what you had in mind for beefing up the unit tests but if you have any suggestions happy to throw them in

### CODE REVIEW

**Comment by dicato** [comment]
@joshblum regarding the tests, if I remember correctly you have three attributes you are performing logic on:

* whether the profile is activated
* whether the user is active
* and, to some extent, whether the user exists

The unit tests did not cover all these situations.

File: registration/models.py [hunk 1]
```diff
@@ -265,12 +266,13 @@ def delete_expired_users(self):
         """
         for profile in self.all():
             try:
-                if profile.activation_key_expired():
+                if profile.activation_key_expired() and not profile.activated:
                     user = profile.user
-                    if not user.is_active:
-                        profile.delete()
-                        user.delete()
+                    logger.warning('Deleting expired Registration profile {} and user {}.'.format(profile, user))
```
Comment (dicato) [inline registration/models.py]: I don't believe `logging` is used anywhere else in this project; how will this impact users of this package? What happens if they don't have logging setup?

Comment (joshblum) [reply registration/models.py]: It's not used elsewhere and I haven't noticed in a few other third party django apps I checked. But is there an alternative? We can suggest a default logging configuration in the docs and mention that it's necessary for the cron job unless there is a different approach we can take.

**Comment by joshblum** [comment]
Awesome, added a test for an active and expired user. Let me know your opinion on adding a logging config to the docs or some other approach!

File: registration/models.py (hunk 1 above)
Comment (dicato) [reply registration/models.py]: I poked around a bit and convinced myself this should be good. :smile:

Other packages follow the same pattern.

Comment (joshblum) [reply registration/models.py]: cool i'll merge in then!

**Pull request merged**